cmd/up/logs.go

```diff
@@ -0,0 +1,142 @@
+// Copyright 2022 The Okteto Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package up
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"os"
+	"os/signal"
+	"regexp"
+	"text/template"
+	"time"
+
+	"github.com/fatih/color"
+	"github.com/okteto/okteto/pkg/config"
+	oktetoErrors "github.com/okteto/okteto/pkg/errors"
+	oktetoLog "github.com/okteto/okteto/pkg/log"
+	"github.com/okteto/okteto/pkg/okteto"
+	"github.com/pkg/errors"
+	"github.com/stern/stern/stern"
+	"k8s.io/apimachinery/pkg/fields"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+var tempKubeConfigTemplate = "%s/.okteto/kubeconfig-%s"
+
+func (up *upContext) showDetachedLogs(ctx context.Context) error {
+	tmpKubeconfig, err := createTempKubeconfig(up.Manifest.Name)
+	if err != nil {
+		return err
+	}
+	defer os.Remove(tmpKubeconfig)
+
+	c, err := getSternConfig(tmpKubeconfig)
+	if err != nil {
+		return err
+	}
+	stop := make(chan os.Signal, 1)
+	signal.Notify(stop, os.Interrupt)
+	exit := make(chan error, 1)
+	go func() {
+		if err := stern.Run(ctx, c); err != nil {
+			exit <- oktetoErrors.UserError{
+				E: fmt.Errorf("failed to get logs: %w", err),
+			}
+		}
+	}()
+	select {
+	case <-stop:
+		oktetoLog.Infof("CTRL+C received, starting shutdown sequence")
+		return oktetoErrors.ErrIntSig
+	case err := <-exit:
+		if err != nil {
+			oktetoLog.Infof("exit signal received due to error: %s", err)
+			return err
+		}
+	}
+	return nil
+}
+
+func createTempKubeconfig(name string) (string, error) {
+	cfg := okteto.Context().Cfg
+	destKubeconfigFile := fmt.Sprintf(tempKubeConfigTemplate, config.GetUserHomeDir(), name)
```

Comment:
is it possible to not create this file? if it's the only option, could we run a `defer` to delete it on exit?